test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Tests/ActionOverloadingEndToEndTests.cs

```diff
@@ -0,0 +1,285 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ActionOverloadingEndToEndTests.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Xml;
+using Microsoft.AspNetCore.OData;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Client.E2E.TestCommon;
+using Microsoft.OData.Client.E2E.Tests.Common.Clients.EndToEnd.Default;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+using Microsoft.OData.Edm.Csdl;
+using Microsoft.OData.Edm.Validation;
+using Microsoft.OData.Edm;
+using Xunit;
+using Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Server;
+
+namespace Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Tests
+{
+    [TestCaseOrderer("Microsoft.OData.Client.E2E.TestCommon.PriorityOrderer", "Microsoft.OData.Client.E2E.TestCommon")]
+    public class ActionOverloadingEndToEndTests : EndToEndTestBase<ActionOverloadingEndToEndTests.TestsStartup>
+    {
+        private readonly Uri _baseUri;
+        private readonly Container _context;
+        public class TestsStartup : TestStartupBase
+        {
+            public override void ConfigureServices(IServiceCollection services)
+            {
+                services.ConfigureControllers(typeof(ProductsController), typeof(OrderLinesController), typeof(PeopleController));
+
+                services.AddControllers().AddOData(opt => opt.Count().Filter().Expand().Select().OrderBy().SetMaxTop(null)
+                    .AddRouteComponents("odata", CommonEndToEndEdmModel.GetEdmModel()));
+            }
+        }
+        public ActionOverloadingEndToEndTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+            _context = new Container(_baseUri);
+            _context.HttpClientFactory = HttpClientFactory;
+        }
+
+        [Fact]
+        public void ExecuteOverloadedOperations()

```

Comment:
What kind of overloads does this test? Calling operations that have the same name but different binding parameters? If this is the entire scope of the test, that maybe the test name should be more specific. Since, for example, this is not testing unbound operations, operations with different parameters (other than the binding parameter), etc.